### PR TITLE
EOS-18065 : Single node cluster on VM using libfabric transport

### PR DESCRIPTION
**Pull request by jamohite**
Signed-off-by: Jaydeep Mohite <jaydeep.mohite@seagate.com>

### CODE REVIEW

**Review by papan-singh — APPROVED** [review]
looks good

**Comment by huanghua78** [comment]
@jamohite please check the branch and commit history. I think you need to rebase the branch.

**Comment by huanghua78** [comment]
Hello @jamohite , can you explain why these changes in  https://github.com/Seagate/cortx-motr/pull/587/commits/5e3b9074895b702a60d27997a39944b4de2acda1 are needed?

**Comment by jamohite** [comment]
> Hello @jamohite , can you explain why these changes in [5e3b907](https://github.com/Seagate/cortx-motr/commit/5e3b9074895b702a60d27997a39944b4de2acda1) are needed?

Changed in UT files are required are due to max segments supported for libfabric for RPC  = 1

File: net/bulk_emulation/mem_xprt_pvt.h [hunk 1]
```diff
@@ -36,9 +36,9 @@
 */
 
 enum {
-	M0_NET_BULK_MEM_MAX_BUFFER_SIZE     = (1 << 19),
-	M0_NET_BULK_MEM_MAX_SEGMENT_SIZE    = (1 << 18),
-	M0_NET_BULK_MEM_MAX_BUFFER_SEGMENTS = 256,
+	M0_NET_BULK_MEM_MAX_BUFFER_SIZE     = (1 << 20),
+	M0_NET_BULK_MEM_MAX_SEGMENT_SIZE    = (1 << 20),
+	M0_NET_BULK_MEM_MAX_BUFFER_SEGMENTS = 1,
```
Comment (huanghua78) [inline net/bulk_emulation/mem_xprt_pvt.h:41]: Please explain the reason of these changes.

File: net/bulk_emulation/ut/bulk_mem_ut.c [hunk 2]
```diff
@@ -298,7 +298,7 @@ static void test_failure(void)
 	M0_UT_ASSERT(d1tm1.ntm_state == M0_NET_TM_STARTED);
 	M0_UT_ASSERT(strcmp(d1tm1.ntm_ep->nep_addr, "127.0.0.1:10") == 0);
 	M0_SET0(&d1nb1);
-	M0_UT_ASSERT(!m0_bufvec_alloc(&d1nb1.nb_buffer, 4, 10));
+	M0_UT_ASSERT(!m0_bufvec_alloc(&d1nb1.nb_buffer, 1, 40));
```
Comment (huanghua78) [inline net/bulk_emulation/ut/bulk_mem_ut.c:301]: Please explain the reason of this change.

File: net/bulk_emulation/ut/bulk_mem_ut.c [hunk 3]
```diff
@@ -675,10 +675,10 @@ static void test_failure(void)
...
 
 enum {
-	PING_CLIENT_SEGMENTS = 8,
-	PING_CLIENT_SEGMENT_SIZE = 512,
-	PING_SERVER_SEGMENTS = 4,
-	PING_SERVER_SEGMENT_SIZE = 1024,
+	PING_CLIENT_SEGMENTS = 1,
+	PING_CLIENT_SEGMENT_SIZE = 4096,
+	PING_SERVER_SEGMENTS = 1,
+	PING_SERVER_SEGMENT_SIZE = 4096,
```
Comment (huanghua78) [inline net/bulk_emulation/ut/bulk_mem_ut.c:681]: Why is this needed?

File: net/bulk_emulation/mem_xprt_pvt.h (hunk 1 above)
Comment (jamohite) [reply net/bulk_emulation/mem_xprt_pvt.h:41]: In case of libfabric, for RPC max seg_nr = 1 and max segment size = 1MB

File: net/bulk_emulation/ut/bulk_mem_ut.c (hunk 2 above)
Comment (jamohite) [reply net/bulk_emulation/ut/bulk_mem_ut.c:301]: In case of libfabric, for RPC max seg_nr = 1 and max segment size = 1MB

File: net/bulk_emulation/ut/bulk_mem_ut.c (hunk 3 above)
Comment (jamohite) [reply net/bulk_emulation/ut/bulk_mem_ut.c:681]: In case of libfabric, for RPC max seg_nr = 1 and max segment size = 1MB
